src/components/visualizations/chart/legend/HeatMapLegend.tsx

```diff
@@ -0,0 +1,93 @@
+import { numberFormat } from '@gooddata/numberjs';
+import get = require('lodash/get');
+import head = require('lodash/head');
+import last = require('lodash/last');
+import * as classNames from 'classnames';
+
+export interface IHeatMapLegendProps {
+    series: any;
+    format?: string;
+}
+
+export function formatLegendLabel(value: number, format: string, diff: number): string {
+    if (format && format.includes('%')) {
+        return numberFormat(value, '#,#0%');
+    }
+
+    const sign = Math.sign(value) === -1 ? '-' : '';
+    const positiveValue = Math.abs(value);
+    let formattingString = '';
+
+    if (diff < 10) {
+        formattingString += '[<1]0.00;[<10]#.#;[<100]#.#;';
+    }
+
+    // tslint:disable-next-line:max-line-length
+    formattingString += '[<1000]0;[<10000]#.#,k;[<999500]#,k;[<10000000]#.#,,M;[<999500000]#,,M;[<10000000000]#.#,,,B;[<999500000000]#,,,B;[<10000000000000]#.#,,,T;[>=10000000000000]#,,,T';
+    return sign + numberFormat(positiveValue, formattingString);
+}
+
+export default class HeatMapLegend extends React.PureComponent<IHeatMapLegendProps> {
+    public render() {
+        const { series, format } = this.props;
+        const min = get(head(series), 'range.from', 0);
+        const max = get(last(series), 'range.to', 0);
+        const diff =  max - min;
+
+        // don't show legend when only one value is present
+        if (this.isSingleValueInSeries(series)) {
+            return null;
+        }
+
+        return (
+            <div className="heatmap-legend">
+                <div className="legend-labels">
+                {
+                    Array.from(Array(series.length + 1)).map((_, index) => {
+                        let value;
+                        let align;
+
+                        if (index === 0) {
+                            value = formatLegendLabel(get(series, '0.range.from', 0), format, diff);
+                            align = 'left';
+                        } else if (index === series.length) {
+                            value = formatLegendLabel(get(series, `${index - 1}.range.to`, 0), format, diff);
+                            align = 'right';
+                        } else {
+                            value = formatLegendLabel(get(series, `${index}.range.from`, 0), format, diff);
+                            align = 'center';
+                        }
+
+                        const classes = classNames('label', align);
+                        return (
+                            <span
+                                className={classes}
+                                key={`label-${index}`}
+                            >
+                                {value}
+                            </span>
+                        );
+                    })
+                }
+                </div>
+                <div className="legend-boxes">
+                    {series.map((item: any, index: number) => {
+                        const style = {
```

Comment:
I'm not sure about this one, but maybe it would be handy to have style creation in a separate function

Reply:
agreed